src/types/integer.rs

```diff
@@ -136,6 +161,90 @@ impl From<UInteger> for Integer {
     }
 }
 
+impl From<BigUint> for UInteger {
+    fn from(value: BigUint) -> Self {
+        // prefer a compact representation for the magnitude
+        match value.to_u64() {
+            Some(unsigned) => UInteger::U64(unsigned),
+            None => UInteger::BigUInt(value),
+        }
+    }
+}
+
+impl From<UInteger> for BigUint {
+    fn from(value: UInteger) -> Self {
+        use UInteger::*;
+        match value {
+            U64(m) => BigUint::from(m),
+            BigUInt(m) => m,
+        }
+    }
+}
+
+impl ToBigUint for UInteger {
+    fn to_biguint(&self) -> Option<BigUint> {
+        // This implementation never fails, but the trait requires an `Option` return type.
+        Some(self.clone().into())
+    }
+}
+
+// This macro makes it possible to turn unsigned int primitives into a Coefficient using `.into()`.
+// Note that it works for both signed and unsigned ints. The resulting UInteger will be the
+// absolute value of the integer being converted.
+macro_rules! impl_uinteger_from_small_unsigned_int_types {
+    ($($t:ty),*) => ($(
+        impl From<$t> for UInteger {
+            fn from(value: $t) -> UInteger {
+                UInteger::U64(value as u64)
+            }
+        }
+    )*)
+}
+
+impl_uinteger_from_small_unsigned_int_types!(u8, u16, u32, u64, usize);
+
+macro_rules! impl_uinteger_from_small_signed_int_types {
+    ($($t:ty),*) => ($(
+        impl From<$t> for UInteger {
+            fn from(value: $t) -> UInteger {
+                // Discard the sign and convert the value to a u64.
+                let magnitude: u64 = value.checked_abs()
+                        .and_then(|v| Some(v.abs() as u64))
```

Comment:
Long ago when I wrote this, Rust didn't have the [`unsigned_abs`](https://doc.rust-lang.org/std/primitive.i64.html#method.unsigned_abs) method to handle this conversion.

Reply:
ack. changed to use that.